### PR TITLE
Goodbye python2

### PR DESCRIPTION
**Pull request by wilko77**
we remove python2 support. It's dead. 

- Removed all the backports. 
- Removed all the python2 testing from the CI
- Removed dependencies only used with python2
- modernized the typing annotations

### CODE REVIEW

File: clkhash/bloomfilter.py [hunk 1]
```diff
@@ -26,13 +24,12 @@
...
-def double_hash_encode_ngrams(ngrams,   # type: Iterable[str]
-                              keys,     # type: Sequence[bytes]
-                              ks,       # type: Sequence[int]
-                              l,        # type: int
-                              encoding  # type: str
-                              ):
-    # type: (...) -> bitarray
+def double_hash_encode_ngrams(ngrams: Iterable[str],
+                              keys: Sequence[bytes],
+                              ks: Sequence[int],
```
Comment (JoyceXinyueWang) [inline clkhash/bloomfilter.py:29]: Nice! I prefer this than the comment style type annotation

File: clkhash/clk.py [hunk 2]
```diff
@@ -85,7 +82,7 @@ def generate_clk_from_csv(input_f,  # type: TextIO
     log.info("Hashing data")
 
     # Read from CSV file
-    reader = unicode_reader(input_f)
+    reader = csv.reader(input_f)
```
Comment (JoyceXinyueWang) [inline clkhash/clk.py:85]: What is the reason for using `csv` over `unicode_reader`?

Comment (wilko77) [reply clkhash/clk.py:85]: unicode_reader was a workaround for python2 to be able to read unicode in csv files. In python3 the csv.reader can do that natively.

File: clkhash/bloomfilter.py (hunk 1 above)
Comment (wilko77) [reply clkhash/bloomfilter.py:29]: me too. Unfortunately, in python 3.5 you cannot yet specify types of variables like this:
`n:int = 10` :(
We still need to use the comment style in those cases

**Pull request merged**